### PR TITLE
helm: Remove redundant capabilities

### LINKED ISSUE #17118: Question: Why NET_ADMIN and SYS_MODULE capabilities added in the securitycontext of  cilium agent container ？
Per the below link, Cilium agent is running in privileged container,  in its securitycontext,  NET_ADMIN and SYS_MODULE capabilities are added,  is it really necessary? I suppose priviliged container has the host's root privilege.

https://github.com/cilium/cilium/blob/b2a5743f3bae0e803acc29d7cdeddfd066dcfa73/install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml#L243

### PR DESCRIPTION
**Pull request by gandro**
The cilium agent container already runs as a privileged container, so
there is no need to grant additional capabilities, as it already has all
of them. This section will become relevant again once we remove
`privileged` from the DaemonSet, but in the meantime there is no reason
to keep an incomplete and redundant list of capabilities.

I did a quick `git blame` on the source of the two capabilities that we had,
and I don't see any rationale as to why they were added. It's also clear
that the set of capabilities is incomplete, as evident by the list
provided in #14446.

I also manually verified this locally, here is the set of capabilities
given to the `cilium-agent` process _with this change applied_,
note `sys_module` and `net_admin` are both present.

```
# pscap -a
ppid  pid   name        command           capabilities
0     1     root        cilium-agent      chown, dac_override, dac_read_search, fowner, fsetid, kill, setgid, setuid, setpcap, linux_immutable, net_bind_service, net_broadcast, net_admin, net_raw, ipc_lock, ipc_owner, sys_module, sys_rawio, sys_chroot, sys_ptrace, sys_pacct, sys_admin, sys_boot, sys_nice, sys_resource, sys_time, sys_tty_config, mknod, lease, audit_write, audit_control, setfcap, mac_override, mac_admin, syslog, wake_alarm, block_suspend, audit_read
```

Fixes: #17118

### CODE REVIEW

**Review by aanm — APPROVED** [review]

**Comment by gandro** [comment]
test-me-please

**Comment by gandro** [comment]
ConformanceEKS is currently known to be broken https://github.com/cilium/cilium/issues/17139

File: install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml [hunk 1]
```diff
@@ -241,10 +241,6 @@ spec:
         {{- end }}
         {{- end }}
         securityContext:
-          capabilities:
-            add:
-            - NET_ADMIN
```
Comment (sayboras) [inline install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml:246]: should these capabilities be cleanup from from other k8s manifest files (e.g. test/k8sT/manifests/*) as well ?

**Review by sayboras — APPROVED** [review]

File: install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml (hunk 1 above)
Comment (gandro) [reply install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml:246]: Good call! I did a quick grep and found some more redundant capabilities. I removed them all.

**Comment by gandro** [comment]
test-me-please

**Review by tklauser — APPROVED** [review]

**Comment by gandro** [comment]
Image build was somehow not triggered. Reopening to kick off GitHub Actions again.

**Review by qmonnet — APPROVED** [review]

**Comment by gandro** [comment]
test-me-please

**Comment by gandro** [comment]
k8s-1.19-kernel-5.4 hit #16852 https://jenkins.cilium.io/job/Cilium-PR-K8s-1.19-kernel-5.4/843/

Some CI 3.0 timed out due to the image build still being wonky due to GitHub. Restarting them.

**Comment by gandro** [comment]
ci-awscni

**Comment by gandro** [comment]
ci-aks

**Comment by gandro** [comment]
ci-eks

**Comment by gandro** [comment]
ci-gke

**Comment by gandro** [comment]
ci-external-workloads

**Comment by gandro** [comment]
ci-multicluster

**Comment by tklauser** [comment]
test-1.19-5.4

**Comment by tklauser** [comment]
ci-gke

**Comment by tklauser** [comment]
ci-eks

**Comment by gandro** [comment]
I'm marking this ready-to-merge. The only CI failure on ConformanceGKE (which doesn't even use the Helm charts, thus is unrelated) failed in `no-policies/pod-to-world/http-to-cilium-io-1`, which is known to be flaky: https://github.com/cilium/cilium/runs/3419400227?check_suite_focus=true

**Comment by dawgwe1** [comment]
Close all issues accounts and accounts 
Then delete all accounts

**Comment by dawgwe1** [comment]
Close all issues accounts and accounts 
Then delete all accounts

**Pull request merged**